Spigot-Server-Patches/0419-Ticket-level-per-player-view-distance-API.patch

```diff
@@ -0,0 +1,1150 @@
+             float pitch = 0.8F + this.random.nextFloat() * 0.2F;
+-            int viewDistance = ((WorldServer) this.world).getServer().getViewDistance() * 16;
++            //int viewDistance = ((WorldServer) this.world).getServer().getViewDistance() * 16;
+             for (EntityPlayer player : (List<EntityPlayer>) (List) this.world.getPlayers()) {
++                int viewDistance = player.getTrackViewDistance() * 16; //Paper ticket level pp-view-dist
+                 double deltaX = this.locX - player.locX;
+                 double deltaZ = this.locZ - player.locZ;
+                 double distanceSquared = deltaX * deltaX + deltaZ * deltaZ;
+diff --git a/src/main/java/net/minecraft/server/EntityWither.java b/src/main/java/net/minecraft/server/EntityWither.java
+index 8b3052b11..dc2ff9482 100644
+--- a/src/main/java/net/minecraft/server/EntityWither.java
++++ b/src/main/java/net/minecraft/server/EntityWither.java
+@@ -208,9 +208,9 @@ public class EntityWither extends EntityMonster implements IRangedEntity {
+                 // CraftBukkit start - Use relative location for far away sounds
+                 // this.world.b(1023, new BlockPosition(this), 0);
+                 // Paper start
+-                int viewDistance = ((WorldServer) this.world).spigotConfig.viewDistance * 16; // Paper - updated to use worlds actual view distance incase we have to uncomment this due to removal of player view distance API
++                //int viewDistance = ((WorldServer) this.world).spigotConfig.viewDistance * 16; // Paper - updated to use worlds actual view distance incase we have to uncomment this due to removal of player view distance API
+                 for (EntityPlayer player : ((WorldServer)world).getPlayers()) {
+-                    //final int viewDistance = player.getViewDistance(); // TODO apply view distance api patch
++                    final int viewDistance = player.getTrackViewDistance() * 16; //Paper ticket level pp-view-dist
+                     // Paper end
+                     double deltaX = this.locX - player.locX;
+                     double deltaZ = this.locZ - player.locZ;
+diff --git a/src/main/java/net/minecraft/server/PlayerChunk.java b/src/main/java/net/minecraft/server/PlayerChunk.java
+index 34d0ab0d5..47592917f 100644
+--- a/src/main/java/net/minecraft/server/PlayerChunk.java
++++ b/src/main/java/net/minecraft/server/PlayerChunk.java
+@@ -103,6 +103,27 @@ public class PlayerChunk {
+         return this.entityTickingFuture;
+     }
+ 
++    //Paper start ticket level pp-view-dist
++    public CompletableFuture<Either<Chunk, PlayerChunk.Failure>> getEntityTickingChunkFuture() { return this.b(); } //Paper OBFHELPER
++    public CompletableFuture<Either<Chunk, PlayerChunk.Failure>> getTickingChunkFuture() { return this.a(); } //Paper OBFHELPER
```

Comment:
obfhelpers go above or on the method declaration
there's no need for these anyways given the underlying fields are mapped

Reply:
solved in rewrite already